src/elements/element.line.js

```diff
@@ -145,35 +326,106 @@ class Line extends Element {
 		super(props);
 	}
 
-	draw(ctx) {
+	set points(points) {
+		this._children = points;
+		delete this._segments;
+	}
+
+	get points() {
+		return this._children;
+	}
+
+	get segments() {
+		return this._segments || (this._segments = computeSegments(this));
+	}
+
+	first() {
+		const segments = this.segments;
+		const points = this.points;
+		return segments.length && points[segments[0].start]._view;
+	}
+
+	last() {
+		const segments = this.segments;
+		const points = this.points;
+		const count = segments.length;
+		return count && points[segments[count - 1].end]._view;
+	}
+
+	interpolate(point, axis) {
 		const me = this;
 		const vm = me._view;
-		const spanGaps = vm.spanGaps;
-		let closePath = me._loop;
-		let points = me._children;
+		const value = point[axis];
+		const points = me.points;
+		const segments = boundSegments(me, {axis, start: value, end: value});
 
-		if (!points.length) {
+		if (!segments.length) {
 			return;
 		}
 
-		if (closePath) {
-			points = startAtGap(points, spanGaps);
-			closePath = points.closePath;
+		const result = [];
+		const method = getInterpolationMethod(vm);
+		let i, ilen;
+		for (i = 0, ilen = segments.length; i < ilen; ++i) {
+			const {start, end} = segments[i];
+			const p1 = points[start]._view;
+			const p2 = points[end]._view;
+			if (p1 === p2) {
+				result.push(p1);
+				continue;
+			}
+			const t = Math.abs((value - p1[axis]) / (p2[axis] - p1[axis]));
+			let interpolated = method(p1, p2, t, vm.steppedLine);
```

Comment:
Can we call this `interpolate` instead of `method`?

Reply:
renamed it to `_interpolate` to make it clear we are not recursively calling `interpolate` that we are in.